server/src/future.rs

```diff
@@ -241,3 +241,43 @@ impl Future for ShutdownWaiter {
 		}
 	}
 }
+
+/// A `Future` that resolves once the server has been requested to be stopped.
+#[derive(Debug)]
+pub(crate) struct ServerStopped(Weak<MonitorInner>);
```

Comment:
Maybe this is a naming thing, but I'm a bit confused about what the difference between `ShutdownWaiter` (which waits for the server to be stopped) and `ServerStopped` (which I think just waits for the shutdown to begin?)

And this comes from calling `stopped`, but it returns a future that's about whether a stop is requested, not whether it has finished stopping?

Perhaps this should be called `ShutdownRequestedWaiter`? to align with `ShutdownWaiter` above? and the `stopped` function should be called `wait_for_shutdown_requested` or something?

Reply:
yepp, this is just a future that revolves when the `server.stop` has been called , so `ShutdownRequestedWaiter` is better.

this is because we need to trigger the `graceful shutdown` of hyper, and shutdown waiter waits for all background tasks to complete.

I need to double-check again why the `ShutdownWaiter` didn't work for this again, I don't remember